tests/ShopBundle/Functional/Model/Product/Search/ProductSearchExportRepositoryTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Tests\ShopBundle\Functional\Model\Product\Search;
+
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportRepository;
+use Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportWithFilterRepository;
+use Tests\ShopBundle\Test\TransactionFunctionalTestCase;
+
+class ProductSearchExportRepositoryTest extends TransactionFunctionalTestCase
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportRepository
+     */
+    private $repository;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    private $domain;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->repository = $this->getContainer()->get(ProductSearchExportRepository::class);
+        $this->domain = $this->getContainer()->get(Domain::class);
+    }
+
+    public function testProductDataHaveExpectedStructure(): void
+    {
+        $data = $this->repository->getProductsData($this->domain->getId(), $this->domain->getLocale(), 0, 10);
+        $this->assertCount(10, $data);
+
+        $structure = array_keys(reset($data));
+        sort($structure);
+
+        $expectedStructure = $this->getExpectedStructureForRepository($this->repository);
+
+        sort($expectedStructure);
+
+        $this->assertSame($expectedStructure, $structure);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportRepository $productSearchExportRepository
+     * @return string[]
+     */
+    private function getExpectedStructureForRepository(ProductSearchExportRepository $productSearchExportRepository): array
+    {
+        $structure = [
+            'id',
+            'name',
+            'catnum',
+            'partno',
+            'ean',
+            'description',
+            'shortDescription',
+        ];
+
+        if ($productSearchExportRepository instanceof ProductSearchExportWithFilterRepository) {
+            $structure = \array_merge($structure, [
```

Comment:
this is strange how `array_keys` method doesn't need global namespace prefix but `array_merge` does

Reply:
We don't have any standards about using functions from the global namespace... This is a copy-pasted file according to our upgrade notes.